main/src/observables/time_energy_growth.hpp

```diff
@@ -30,78 +30,74 @@
  * @author Lukas Schmidt
  */
 
+#include <array>
+#include <mpi.h>
+
 #include "iobservables.hpp"
 #include "sph/math.hpp"
-#include <mpi.h>
 #include "io/ifile_writer.hpp"
 
+namespace sphexa
+{
 
-namespace sphexa {
+template<class Tc, class Tm>
+std::array<Tc, 3> localGrowthRate(size_t startIndex, size_t endIndex, const Tc* x, const Tc* y, const Tm* vy,
+                                  const Tm* rho, const Tm* m, const Tm* kx, const cstone::Box<Tc>& box)
+{
+    const Tc ybox = box.ly();
 
+    Tc sumsi = 0;
+    Tc sumci = 0;
+    Tc sumdi = 0;
 
-template<typename T, class Dataset>
-void localGrowthRate(size_t startIndex, size_t endIndex, Dataset& d, T* sumsi, T* sumci, T* sumdi, const cstone::Box<T>& box)
-{
-    const T* x  =   d.x.data();
-    const T* y  =   d.y.data();
-    const T* vy =   d.vy.data();
-    const T* rho =  d.rho.data();
-    const T* m =    d.m.data();
-    const T* kx =   d.kx.data();
-    const T ybox =  box.ly();
-
-    T sumsiThread = 0.0, sumciThread = 0.0, sumdiThread = 0.0;
```

Comment:
These variables are defined outside the OpenMP clause and are therefore not thread-local.
OpenMP will automatically generate thread-local variables for the reduction.